src/status_im/protocol/handlers.cljs

```diff
@@ -1,291 +1,204 @@
             [status-im.data-store.messages :as messages]
             [status-im.data-store.pending-messages :as pending-messages]
             [status-im.data-store.processed-messages :as processed-messages]
             [status-im.data-store.chats :as chats]
             [status-im.protocol.core :as protocol]
-            [status-im.constants :refer [text-content-type
-                                         blocks-per-hour]]
-            [status-im.i18n :refer [label]]
+            [status-im.constants :as constants]
+            [status-im.i18n :as i18n]
             [status-im.utils.random :as random]
             [status-im.protocol.message-cache :as cache]
-            [status-im.utils.datetime :as dt]
+            [status-im.utils.datetime :as datetime]
             [taoensso.timbre :as log :refer-macros [debug]]
-            [status-im.constants :as c]
             [status-im.native-module.core :as status]
-            [clojure.string :refer [join]]
-            [status-im.utils.scheduler :as s]
-            [status-im.utils.web3-provider :as w3]
-            [status-im.utils.datetime :as time]))
+            [clojure.string :as string]
+            [status-im.utils.web3-provider :as web3-provider]))
+
+;;;; COFX
+
+(re-frame/reg-cofx
+  ::get-web3
+  (fn [coeffects _]
+    (assoc coeffects :web3 (web3-provider/make-web3))))
+
+(re-frame/reg-cofx
+  ::get-chat-groups
+  (fn [coeffects _]
+    (assoc coeffects :groups (chats/get-active-group-chats))))
+
+(re-frame/reg-cofx
+  ::get-pending-messages
+  (fn [coeffects _]
+    (assoc coeffects :pending-messages (pending-messages/get-all))))
+
+(re-frame/reg-cofx
+  ::get-all-contacts
+  (fn [coeffects _]
+    (assoc coeffects :contacts (contacts/get-all))))
+
+(re-frame/reg-cofx
+  ::message-get-by-id
+  (fn [coeffects _]
+    (let [[{{:keys [message-id]} :payload}] (:event coeffects)]
+      (assoc coeffects :message-by-id (messages/get-by-id message-id)))))
+
+(re-frame/reg-cofx
+  ::chats-new-update?
+  (fn [coeffects _]
+    (let [[{{:keys [group-id timestamp]} :payload}] (:event coeffects)]
+      (assoc coeffects :new-update? (chats/new-update? timestamp group-id)))))
+
+(re-frame/reg-cofx
+  ::chats-is-active-and-timestamp
+  (fn [coeffects _]
+    (let [[{{:keys [group-id timestamp]} :payload}] (:event coeffects)]
+      (assoc coeffects :chats-is-active-and-timestamp
+                       (and (chats/is-active? group-id)
+                            (> timestamp (chats/get-property group-id :timestamp)))))))
+
+(re-frame/reg-cofx
+  ::chats-is-active?
+  (fn [coeffects _]
+    (let [[{{:keys [group-id]} :payload from :from}] (:event coeffects)]
+      (assoc coeffects :chats-is-active?
+                       (chats/is-active? (or group-id from))))))
+
+(re-frame/reg-cofx
+  ::has-contact?
+  (fn [coeffects _]
+    (let [[{{:keys [group-id identity]} :payload}] (:event coeffects)]
+      (assoc coeffects :has-contact? (chats/has-contact? group-id identity)))))
```

Comment:
I know that it was probably not the intention of this PR, but while we are doing this, we should get rid of all those (above 4) bizarre co-effects which could be just plain simple functions taking app-db (or part of it) as argument.
Unless there is some urgency to having protocols refactored with `cofx`/`fx` ASAP (tell me if yes), I would rather refactor it properly all the way down, instead of of just pushing it in front of us.
We plan have plan to revise the realm usage (probably even drop/replace it) anyway, so the more unnecessary realm access we remove, the better.

Reply:
totally agree with you. yes, i'm working on desktop and just need cofx/fx ASAP, decided to just move this bizarre to co-effects, and make cofx/fx "cleaning" later